doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+accepts a `random_state` parameter::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split
+
+    >>> rng = 42  # or any other integer
+    >>> X, y = make_classification(random_state=rng)
+    >>> rf = RandomForestClassifier(random_state=rng)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ...                                                     random_state=rng)
+    >>> rf.fit(X_train, y_train).score(X_test, y_test)
+    0.88
+
+We are now guaranteed that the result of this script will always be 0.88, no
+matter how many times we run it. Changing the global `rng` variable to a
+different value should affect the results, as expected.
+
+We could have used a `RandomState` instance instead of an integer::
+
+    >>> import numpy as np
+    >>> rng = np.random.RandomState(0)
+
+This too would give us reproducible results across executions. However, using
+`RandomState` instances may have surprising and unwanted effects that will be
+described later.
+
+We do not recommend either setting the global `numpy` seed by calling
+`np.random.seed(0)`. See `here
+<https://stackoverflow.com/questions/5836335/consistently-create-same-random-numpy-array/5837352#comment6712034_5837352>`_
+for a discussion.
+
+Using None or `RandomState` instances
+-------------------------------------
+
+Estimators
+..........
+
+As described above, passing instances means that calling `fit` multiple
+times will not yield the same results, even if the estimator is fitted on the
+same data, with the same hyper-parameters::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split
+    >>> import numpy as np
+
+    >>> rng = np.random.RandomState(0)
+    >>> X, y = make_classification(random_state=rng)
+    >>> rf = RandomForestClassifier(max_features=2, max_samples=10,
+    ...                             random_state=rng)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ...                                                     random_state=rng)
+
+    >>> rf.fit(X_train, y_train).score(X_test, y_test)
+    0.76
+    >>> rf.fit(X_train, y_train).score(X_test, y_test)
+    0.8
+
+.. note::
+    `score()` is not a random procedure. Only `fit` has randomness.
+
+We can see from the snippet above that `rf.fit` has produced different
+models, even if the data was the same. This is because the RNG of the
+estimator is consumed when `fit` is called, and this consumed (mutaded) RNG
+will be used in the subsequent calls to `fit`. In addition, the `rng`
+object is shared across any object that uses it, and as a consequence, these
+object become somewhat inter-dependent. For example, two estimator that share
+the same `RandomState` instance will influence each-other, as we will see
```

Comment:
I think this should be "estimators" (plural), and remove the dash from "each other".